crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -246,39 +246,34 @@ pub(crate) fn do_finalize_operator_epoch_staking<T: Config>(
         return Ok((operator.current_total_stake, false));
     }
 
-    let total_stake = operator.current_total_stake;
-    let total_shares = operator.current_total_shares;
+    let mut total_stake = operator.current_total_stake;
+    let mut total_shares = operator.current_total_shares;
     let share_price = SharePrice::new::<T>(total_shares, total_stake);
 
     // calculate and subtract total withdrew shares from previous epoch
-    let (total_stake, total_shares) = if !operator.withdrawals_in_epoch.is_zero() {
```

Comment:
I remeber @nazar-pc mentioning about preferring immutable vars over mutable one. easy to reason with IMHO